### ISSUE TITLE
Unable to define multiple END states via JavaConfig

### ISSUE THREAD

**Issue by heffebaycay** [spring-projects/spring-statemachine#368]
Spring State Machine does not appear to let me define multiple END states when configuring the State Machine workflow using JavaConfig.

Doing this:

```java
    @Override
    public void configure(StateMachineStateConfigurer<WorkflowState,WorkflowEvent> states) throws Exception {
        // STATE configuration
        states.withStates()
                .initial(WorkflowState.INITIAL)
// more configuration
                .end(WorkflowState.CANCELLED)
                .end(WorkflowState.SUCCESS);
    }
```

would only define `WorkflowState.SUCCESS` as an end state. Looking at the implementation (and the interface comments) of the `end()` method, it is clear that it wasn't written with the intent of support multiple END states:

```java
@Override
public StateConfigurer<S, E> end(S end) {
	this.end = end;
	state(end);
	return this;
}
```

However when I look at the OrderShipping sample, I can see that at least two END states are defined and the StateMachine seems to handle this fine. The only difference is OrderShipping is using UML to configure its workflow.

Is it an oversight in the builder?

**Comment by jvalkeal** [comment]
Yeah, don't see any reason not to do this.

**Comment by jvalkeal** [comment]
Merged into 1.2.x per https://github.com/spring-projects/spring-statemachine/commit/78e3e479a7d1b678ddca34b796622d68e716646d
